ffs/manager/manager.go

```diff
@@ -135,6 +133,8 @@ func (m *Manager) Create(ctx context.Context) (ffs.APIID, string, error) {
 	if err != nil {
 		return ffs.EmptyInstanceID, "", fmt.Errorf("generating auth token for %s: %s", fapi.ID(), err)
 	}
+	m.lock.Lock()
+	defer m.lock.Unlock()
```

Comment:
Move the lock closer to protecting the concurrent write in the map. Before it included the API call to lotus which can have deals, making concurrent users creating wallets wait too much time.